In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
titanic = sns.load_dataset('titanic')

In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


# Titanic 정보
```
#   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
```
대부분 891개지만 age를 보면 714개만 있다 이건 NaN 값이 있다는 것

In [ ]:
titanic.describe()

,survived,pclass,age,sibsp,parch,fare
count,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [ ]:
df = titanic.loc[:, ['age', 'fare']]

In [ ]:
df

,age,fare
0,22.0,7.2500
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500
...,...,...
886,27.0,13.0000
887,19.0,30.0000
888,NaN,23.4500
889,26.0,30.0000


In [ ]:
df.head()

,age,fare
0,22.0,7.2500
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500


In [ ]:
df.tail(10)

,age,fare
881,33.0,7.8958
882,22.0,10.5167
883,28.0,10.5000
884,25.0,7.0500
885,39.0,29.1250
886,27.0,13.0000
887,19.0,30.0000
888,NaN,23.4500
889,26.0,30.0000
890,32.0,7.7500


In [ ]:
addi = df + 10

In [ ]:
addi

,age,fare
0,32.0,17.2500
1,48.0,81.2833
2,36.0,17.9250
3,45.0,63.1000
4,45.0,18.0500
...,...,...
886,37.0,23.0000
887,29.0,40.0000
888,NaN,33.4500
889,36.0,40.0000


In [ ]:
subt = addi - df

In [ ]:
subt

,age,fare
0,10.0,10.0
1,10.0,10.0
2,10.0,10.0
3,10.0,10.0
4,10.0,10.0
...,...,...
886,10.0,10.0
887,10.0,10.0
888,NaN,10.0
889,10.0,10.0


In [ ]:
sample1 = addi.tail()

In [ ]:
sample1

,age,fare
886,37.0,23.00
887,29.0,40.00
888,NaN,33.45
889,36.0,40.00
890,42.0,17.75


In [ ]:
# sample2 = subt.tail().fillna(0, 0)

In [ ]:
# sample1.div(sample2, fill_value=0).fillna(0, 0)

In [ ]:
df = df.iloc[:9]

In [ ]:
df

,age,fare
0,22.0,7.2500
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500
5,NaN,8.4583
6,54.0,51.8625
7,2.0,21.0750
8,27.0,11.1333


In [ ]:
df['age']<20

,age
0,False
1,False
2,False
3,False
4,False
5,False
6,False
7,True
8,False


In [ ]:
df[df['age']<20]

,age,fare
7,2.0,21.075


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   age     8 non-null      float64
 1   fare    9 non-null      float64
dtypes: float64(2)
memory usage: 276.0 bytes


In [ ]:
df.reset_index()

,index,age,fare
0,0,22.0,7.2500
1,1,38.0,71.2833
2,2,26.0,7.9250
3,3,35.0,53.1000
4,4,35.0,8.0500
5,5,NaN,8.4583
6,6,54.0,51.8625
7,7,2.0,21.0750
8,8,27.0,11.1333


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   age     8 non-null      float64
 1   fare    9 non-null      float64
dtypes: float64(2)
memory usage: 276.0 bytes


In [ ]:
titanic.loc[0:9, ['age', 'fare']]

,age,fare
0,22.0,7.2500
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500
5,NaN,8.4583
6,54.0,51.8625
7,2.0,21.0750
8,27.0,11.1333
9,14.0,30.0708


In [ ]:
titanic.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [ ]:
df.loc[df['age']<20]

,age,fare
7,2.0,21.075


titanic 자료에서, 10에서 19세 승객 자료만 추출하기

In [ ]:
titanic

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True
887,1,1,female,19.0,0,0,30.0000,S,First,woman,False,B,Southampton,yes,True
888,0,3,female,NaN,1,2,23.4500,S,Third,woman,False,NaN,Southampton,no,False
889,1,1,male,26.0,0,0,30.0000,C,First,man,True,C,Cherbourg,yes,True


In [ ]:
titanic['age']

,age
0,22.0
1,38.0
2,26.0
3,35.0
4,35.0
...,...
886,27.0
887,19.0
888,NaN
889,26.0


In [ ]:
titanic.loc[(titanic['age']>=10) & (titanic['age']<20)]

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
14,0,3,female,14.0,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
22,1,3,female,15.0,0,0,8.0292,Q,Third,child,False,NaN,Queenstown,yes,True
27,0,1,male,19.0,3,2,263.0000,S,First,man,True,C,Southampton,no,False
38,0,3,female,18.0,2,0,18.0000,S,Third,woman,False,NaN,Southampton,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
853,1,1,female,16.0,0,1,39.4000,S,First,woman,False,D,Southampton,yes,False
855,1,3,female,18.0,0,1,9.3500,S,Third,woman,False,NaN,Southampton,yes,False
875,1,3,female,15.0,0,0,7.2250,C,Third,child,False,NaN,Cherbourg,yes,True
877,0,3,male,19.0,0,0,7.8958,S,Third,man,True,NaN,Southampton,no,True


In [ ]:
df_19

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
7,0,3,male,2.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
10,1,3,female,4.0,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
14,0,3,female,14.0,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
16,0,3,male,2.0,4,1,29.1250,Q,Third,child,False,NaN,Queenstown,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
855,1,3,female,18.0,0,1,9.3500,S,Third,woman,False,NaN,Southampton,yes,False
869,1,3,male,4.0,1,1,11.1333,S,Third,child,False,NaN,Southampton,yes,False
875,1,3,female,15.0,0,0,7.2250,C,Third,child,False,NaN,Cherbourg,yes,True
877,0,3,male,19.0,0,0,7.8958,S,Third,man,True,NaN,Southampton,no,True


In [ ]:
# 강사님 버전
mask = (titanic['age']>=10) & (titanic['age']<20)
titanic.loc[mask].head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
14,0,3,female,14.0,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
22,1,3,female,15.0,0,0,8.0292,Q,Third,child,False,NaN,Queenstown,yes,True
27,0,1,male,19.0,3,2,263.0000,S,First,man,True,C,Southampton,no,False
38,0,3,female,18.0,2,0,18.0000,S,Third,woman,False,NaN,Southampton,no,False


titanic 자료에서, 9세미만, 여성 승객자료만 추출

In [ ]:
titanic_female_n = (titanic['sex']=='female') & (titanic['age'] < 9)

In [ ]:
titanic_female_n

,0
0,False
1,False
2,False
3,False
4,False
...,...
886,False
887,False
888,False
889,False


In [ ]:
titanic.loc[titanic_female_n]

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
10,1,3,female,4.00,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
24,0,3,female,8.00,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
43,1,2,female,3.00,1,2,41.5792,C,Second,child,False,NaN,Cherbourg,yes,False
58,1,2,female,5.00,1,2,27.7500,S,Second,child,False,NaN,Southampton,yes,False
119,0,3,female,2.00,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False
172,1,3,female,1.00,1,1,11.1333,S,Third,child,False,NaN,Southampton,yes,False
184,1,3,female,4.00,0,2,22.0250,S,Third,child,False,NaN,Southampton,yes,False
205,0,3,female,2.00,0,1,10.4625,S,Third,child,False,G,Southampton,no,False
233,1,3,female,5.00,4,2,31.3875,S,Third,child,False,NaN,Southampton,yes,False
237,1,2,female,8.00,0,2,26.2500,S,Second,child,False,NaN,Southampton,yes,False


In [ ]:
titanic_n = titanic.loc[titanic['age'] < 9]

In [ ]:
titanic_n.loc[titanic_n['sex'] == 'female']

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
10,1,3,female,4.00,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
24,0,3,female,8.00,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
43,1,2,female,3.00,1,2,41.5792,C,Second,child,False,NaN,Cherbourg,yes,False
58,1,2,female,5.00,1,2,27.7500,S,Second,child,False,NaN,Southampton,yes,False
119,0,3,female,2.00,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False
172,1,3,female,1.00,1,1,11.1333,S,Third,child,False,NaN,Southampton,yes,False
184,1,3,female,4.00,0,2,22.0250,S,Third,child,False,NaN,Southampton,yes,False
205,0,3,female,2.00,0,1,10.4625,S,Third,child,False,G,Southampton,no,False
233,1,3,female,5.00,4,2,31.3875,S,Third,child,False,NaN,Southampton,yes,False
237,1,2,female,8.00,0,2,26.2500,S,Second,child,False,NaN,Southampton,yes,False


In [ ]:
df.query('age < 20')

,age,fare
7,2.0,21.075


In [ ]:
df.query('age >= 20 and fare < 15')

,age,fare
0,22.0,7.2500
2,26.0,7.9250
4,35.0,8.0500
8,27.0,11.1333


In [ ]:
df.query('age not in [21, 22, 23, 24]')

,age,fare
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500
5,NaN,8.4583
6,54.0,51.8625
7,2.0,21.0750
8,27.0,11.1333


In [ ]:
def max_user(x, y):
    return max(x, y)

In [ ]:
max_user(1 ,4)

4

In [ ]:
max(1, 4)

4

In [ ]:
df.query('age > @max_user(1, 10)')

,age,fare
0,22.0,7.2500
1,38.0,71.2833
2,26.0,7.9250
3,35.0,53.1000
4,35.0,8.0500
6,54.0,51.8625
8,27.0,11.1333


In [ ]:
df.query('index > 3')

,age,fare
4,35.0,8.0500
5,NaN,8.4583
6,54.0,51.8625
7,2.0,21.0750
8,27.0,11.1333


In [ ]:
titanic.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [ ]:
titanic['class'].str.match('c') # 시리즈인데 문자열 기능을 붙이려고 str을 붙인다

,class
0,False
1,False
2,False
3,False
4,False
...,...
886,False
887,False
888,False
889,False


In [ ]:
mask1 = titanic['embark_town'] == 'Southampton'
mask2 = titanic['embark_town'].str.match('Queenstown')
df_bool = titanic.loc[mask1 | mask2, :]
df_bool.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True


In [ ]:
isin_filter = titanic['embark_town'].isin(['Queenstown', 'Southampton'])
df_isin = titanic.loc[isin_filter]

In [ ]:
df_isin.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True


In [ ]:
ser = pd.Series(['Apple_사과', 'Banana_바나나', 'Cherry_체리', np.nan],
                index=['First', ' Second', ' Third', 'Fourth'])

In [ ]:
ser

,0
First,Apple_사과
Second,Banana_바나나
Third,Cherry_체리
Fourth,NaN


In [ ]:
ser.str.len()

,0
First,8.0
Second,10.0
Third,9.0
Fourth,NaN


In [ ]:
ser.str.split('_', expand=True) # expand 하면 데이터 프레임으로 만들어진다?

,0,1
First,Apple,사과
Second,Banana,바나나
Third,Cherry,체리
Fourth,NaN,NaN


In [ ]:
ser.index

Index(['First', ' Second', ' Third', 'Fourth'], dtype='object')

In [ ]:
ser.index.str.strip()

Index(['First', 'Second', 'Third', 'Fourth'], dtype='object')

In [ ]:
ser.index.str.replace('T', 'S') # T를 S로

Index(['First', ' Second', ' Shird', 'Fourth'], dtype='object')

In [ ]:
%%writefile ex1.csv
a,b,c,d,message
1,2,3,4,hello
5,6,7,8,world
9,10,11,12,foo

Overwriting ex1.csv


In [ ]:
df = pd.read_csv('ex1.csv')

In [ ]:
df

,a,b,c,d,message
0,1,2,3,4,hello
1,5,6,7,8,world
2,9,10,11,12,foo


In [ ]:
pd.read_table('ex1.csv', sep=',') # sep 구분자

,a,b,c,d,message
0,1,2,3,4,hello
1,5,6,7,8,world
2,9,10,11,12,foo


In [ ]:
pd.read_csv('ex1.csv', header=[0])

,a,b,c,d,message
0,1,2,3,4,hello
1,5,6,7,8,world
2,9,10,11,12,foo


In [ ]:
names=['one', 'two', 'three', 'four', 'message']
pd.read_csv('ex1.csv', names=names, index_col='message')

,one,two,three,four
message,,,,
message,a,b,c,d
hello,1,2,3,4
world,5,6,7,8
foo,9,10,11,12


In [ ]:
%%writefile csv_mindex.csv
key1,key2,value1,value2
one,a,1,2
one,b,3,4
one,c,5,6
one,d,7,8
two,a,9,10
two,b,11,12
two,c,13,14
two,d,15,16

Overwriting csv_mindex.csv


In [ ]:
df = pd.read_csv('csv_mindex.csv', index_col=['key1', 'key2'])

In [ ]:
df

value1  value2
key1 key2                
one  a          1       2
     b          3       4
     c          5       6
     d          7       8
two  a          9      10
     b         11      12
     c         13      14
     d         15      16

In [ ]:
%%writefile ex3
            A         B         C
aaa -0.264438 -1.026059 -0.619500
bbb  0.927272  0.302904 -0.032399
ccc -0.264273 -0.386314 -0.217601
ddd -0.871858 -0.348382  1.100491

Overwriting ex3


In [ ]:
pd.read_table('ex3', sep='\s+') # \s+ : 빈칸이 한개도 되고 여러개도 된다는 뜻

<>:1: SyntaxWarning: invalid escape sequence '\s'
<>:1: SyntaxWarning: invalid escape sequence '\s'
/tmp/ipykernel_7354/4038723380.py:1: SyntaxWarning: invalid escape sequence '\s'
  pd.read_table('ex3', sep='\s+') # \s+ : 빈칸이 한개도 되고 여러개도 된다는 뜻


,A,B,C
aaa,-0.264438,-1.026059,-0.619500
bbb,0.927272,0.302904,-0.032399
ccc,-0.264273,-0.386314,-0.217601
ddd,-0.871858,-0.348382,1.100491


In [ ]:
df.loc[('one', 'a'), :]

,one
,a
value1,1
value2,2


In [ ]:
%%writefile ex4.csv
# hey!
a,b,c,d,message
# just wanted to make things more difficult for you
# who reads CSV files with computers, anyway?
1,2,3,4,hello
5,6,7,8,world
9,10,11,12,foo

Overwriting ex4.csv


In [ ]:
pd.read_csv('ex4.csv', sep=',', skiprows=[0, 2, 3])

,a,b,c,d,message
0,1,2,3,4,hello
1,5,6,7,8,world
2,9,10,11,12,foo


In [ ]:
%%writefile ex5.csv
something,a,b,c,d,message
one,1,2,3,4,NA
two,5,6,7,8,world
three,9,10,11,12,foo

Overwriting ex5.csv


In [ ]:
df = pd.read_csv('ex5.csv', nrows=2)

In [ ]:
df

,something,a,b,c,d,message
0,one,1,2,3,4,NaN
1,two,5,6,7,8,world


In [ ]:
df.to_csv('ex6_.csv')

In [ ]:
!cat ex6_.csv

,something,a,b,c,d,message
0,one,1,2,3,4,
1,two,5,6,7,8,world


In [ ]:
df.to_csv('ex7_.csv', sep='|')

In [ ]:
!cat ex7_.csv

|something|a|b|c|d|message
0|one|1|2|3|4|
1|two|5|6|7|8|world


In [ ]:
df.to_csv('ex8_.csv', sep='|', na_rep='NA')

In [ ]:
!cat ex8_.csv

|something|a|b|c|d|message
0|one|1|2|3|4|NA
1|two|5|6|7|8|world


In [ ]:
df.to_csv('ex9_.csv', index=False)

In [ ]:
!cat ex9_.csv

something,a,b,c,d,message
one,1,2,3,4,
two,5,6,7,8,world


In [ ]:
import sys
df.to_csv(sys.stdout, sep='|')

|something|a|b|c|d|message
0|one|1|2|3|4|
1|two|5|6|7|8|world


In [ ]:
dates = pd.date_range('1/1/2026', periods = 7)
ts = pd.Series(np.arange(7), index=dates)
ts.to_csv('t_series.csv')

In [ ]:
dates

DatetimeIndex(['2026-01-01', '2026-01-02', '2026-01-03', '2026-01-04',
               '2026-01-05', '2026-01-06', '2026-01-07'],
              dtype='datetime64[ns]', freq='D')

In [ ]:
!cat t_series.csv

,0
2026-01-01,0
2026-01-02,1
2026-01-03,2
2026-01-04,3
2026-01-05,4
2026-01-06,5
2026-01-07,6


In [ ]:
ts

,0
2026-01-01,0
2026-01-02,1
2026-01-03,2
2026-01-04,3
2026-01-05,4
2026-01-06,5
2026-01-07,6


In [ ]:
%%writefile ex9.csv
"a","b","c"
"1","2","3"
"1","2","3"

Overwriting ex9.csv


In [ ]:
ls

csv_mindex.csv  ex4.csv  ex6_.csv  ex9.csv       t_series.csv
ex1.csv         ex5.csv  ex7_.csv  ex9_.csv
ex3             ex6.csv  ex8_.csv  sample_data/


In [ ]:
import csv

In [ ]:
f=open('ex9.csv')
reader = csv.reader(f)

In [ ]:
reader

In [ ]:
for line in reader:
    print(line)

['a', 'b', 'c']
['1', '2', '3']
['1', '2', '3']


In [ ]:
f.close()

In [ ]:
with open('ex9.csv') as f:
    lines = list(csv.reader(f))

In [ ]:
lines

[['a', 'b', 'c'], ['1', '2', '3'], ['1', '2', '3']]

In [ ]:
for i in lines:
    print(i)

['a', 'b', 'c']
['1', '2', '3']
['1', '2', '3']


In [ ]:
obj = """
{"name": "Wes",
 "places_lived": ["United States", "Spain", "Germany"],
 "pet": null,
 "siblings": [{"name": "Scott", "age": 30, "pets": ["Zeus", "Zuko"]},
              {"name": "Katie", "age": 38,
               "pets": ["Sixes", "Stache", "Cisco"]}]
}
"""

In [ ]:
obj

'\n{"name": "Wes",\n "places_lived": ["United States", "Spain", "Germany"],\n "pet": null,\n "siblings": [{"name": "Scott", "age": 30, "pets": ["Zeus", "Zuko"]},\n              {"name": "Katie", "age": 38,\n               "pets": ["Sixes", "Stache", "Cisco"]}]\n}\n'

In [ ]:
import json
result = json.loads(obj)
result

{'name': 'Wes',
 'places_lived': ['United States', 'Spain', 'Germany'],
 'pet': None,
 'siblings': [{'name': 'Scott', 'age': 30, 'pets': ['Zeus', 'Zuko']},
  {'name': 'Katie', 'age': 38, 'pets': ['Sixes', 'Stache', 'Cisco']}]}

In [ ]:
result

{'name': 'Wes',
 'places_lived': ['United States', 'Spain', 'Germany'],
 'pet': None,
 'siblings': [{'name': 'Scott', 'age': 30, 'pets': ['Zeus', 'Zuko']},
  {'name': 'Katie', 'age': 38, 'pets': ['Sixes', 'Stache', 'Cisco']}]}

In [ ]:
type(result)

dict

In [ ]:
asjson = json.dumps(result) # json 형태로
asjson

'{"name": "Wes", "places_lived": ["United States", "Spain", "Germany"], "pet": null, "siblings": [{"name": "Scott", "age": 30, "pets": ["Zeus", "Zuko"]}, {"name": "Katie", "age": 38, "pets": ["Sixes", "Stache", "Cisco"]}]}'

In [ ]:
type(asjson)

str

In [ ]:
print(json.dumps({'4':5, '6':7, '8':9}, sort_keys=True, indent=4)) # indent가 뭘까

{
    "4": 5,
    "6": 7,
    "8": 9
}
